crane/views/v2.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import absolute_import
+from flask import Blueprint, json, current_app, redirect
+
+from crane import app_util
+from crane.api import repository
+
+
+section = Blueprint('v2', __name__, url_prefix='/v2')
+
+
+@section.after_request
+def add_common_headers(response):
+    """
+    Add headers to a response.
+
+    All 200 responses get a content type of 'application/json', and all others
+    retain their default.
+
+    Headers are added to make this app look like the actual docker-registry.
+
+    :param response:    flask response object for a request
+    :type  response:    flask.Response
+
+    :return:    a response object that has the correct headers
+    :rtype:     flask.Response
+    """
+    # if response code is 200, assume it is JSON
+    if response.status_code == 200:
+        response.headers['Content-Type'] = 'application/json'
+        response.headers['Docker-Distribution-API-Version'] = 'registry/2.0'
+    return response
+
+
+@section.route('/')
+def v2():
+    """
+    Provides version support information for /v2 requests.
+
+    :return: Empty JSON document in the response body
+    :rtype:  flask.response
+    """
+    # "{}" is what the real docker-registry puts in the response body
+    response = current_app.make_response(json.dumps({}))
+    return response
+
+
+@section.route('/<path:username>/<path:repo>/<path:file_path>')
```

Comment:
After implementing the changes, the curl you have provided gives the following result
[randalap@localhost crane]$ curl -v -X GET http://127.0.0.1:5001/v2/registry/manifests/2
*   Trying 127.0.0.1...
* Connected to 127.0.0.1 (127.0.0.1) port 5001 (#0)
> GET /v2/registry/manifests/2 HTTP/1.1
> User-Agent: curl/7.40.0
> Host: 127.0.0.1:5001
> Accept: */*
> 
* HTTP 1.0, assume close after body
< HTTP/1.0 302 FOUND
< Content-Type: text/html; charset=utf-8
< Content-Length: 301
< Location: https://dev/pulp/docker/v2/registry/manifests/2
< Server: Werkzeug/0.9.6 Python/2.7.10
< Date: Wed, 23 Sep 2015 05:40:04 GMT
< 
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>Redirecting...</title>
<h1>Redirecting...</h1>
* Closing connection 0
<p>You should be redirected automatically to target URL: <a href="https://dev/pulp/docker/v2/registry/manifests/2">https://dev/pulp/docker/v2/registry/manifests/2</a>.  If not click the link.